src/store/actions/updateBalances.ts

```diff
@@ -1,88 +1,110 @@
 import { Address } from '@chainify/types';
-import { ChainId } from '@liquality/cryptoassets';
+import { ChainId, chains } from '@liquality/cryptoassets';
 import Bluebird from 'bluebird';
+import { chunk } from 'lodash';
 import { ActionContext, rootActionContext } from '..';
 import { assetsAdapter } from '../../utils/chainify';
-import { Asset, Network } from '../types';
+import { AccountId, Network, WalletId } from '../types';
 
-export const updateBalances = async (
-  context: ActionContext,
-  { network, walletId, assets }: { network: Network; walletId: string; assets: Asset[] }
-): Promise<void> => {
+type UpdateBalanceRequestType = {
+  walletId: WalletId;
+  network: Network;
+  accountIds?: AccountId[];
+};
+
+export const updateBalances = async (context: ActionContext, request: UpdateBalanceRequestType) => {
+  const { walletId, network, accountIds } = request;
   const { state, commit, getters } = rootActionContext(context);
+  const getClient = getters.client;
 
-  // filter accounts for all enabled assets
-  let accounts = state.accounts[walletId]?.[network].filter((a) => a.assets && a.assets.length > 0 && a.enabled);
+  const allEnabledAccountIds =
```

Comment:
Why calculate all enabled if we're not going to use it (when accoundIds is provided)